git-branchless/src/commands/hide.rs

```diff
@@ -89,14 +113,62 @@ pub fn hide(effects: &Effects, hashes: Vec<String>, recursive: bool) -> eyre::Re
         }
     }
 
+    if delete_branches {
+        // Delete the abandonded branches by "moving" them from their current OID to a Zero OID.
+        let abandoned_branches = abandoned_branches
+            .values()
+            .map(|oid| (*oid, MaybeZeroOid::Zero))
+            .collect();
+        move_branches(
+            effects,
+            git_run_info,
+            &repo,
+            event_tx_id,
+            &abandoned_branches,
+        )?;
+    }
+
+    let abandoned_branches: Vec<&str> = abandoned_branches.into_keys().collect();
```

Comment:
As per above, move the construction of branches from above to down here, maybe something like this:

```
commits
    .iter()
    .flat_map(|commit| branch_oid_to_names.get(commit.get_oid()).unwrap_or_default())
    .collect();
```

You should also sort the branch names before rendering for determinism.

To render the branch name, do something like this rather than manually stripping the suffix above:

```
CategorizedReferenceName::new(branch_name).render_suffix()
```

Reply:
Yup, this is great, thank you! It really tightened up the changes and cleaned things up nicely.